src/globalComponents/apiCalls/marketPlaceAPIConnections.js

```diff
@@ -0,0 +1,142 @@
+//@flow strict
+
+import SaleObject from '../typesAndClasses/SaleObject.js';
+
+/*
+	This function takes the following input
+		userId: the id of the user
+		sellingPrice: how much the user is asking for the bundle of items 
+		itemId: the id of either the tank or the item
+		itemType: either tank or component which determines what this item is 
+		amountOfSellingItems: how many of these items we are selling
+	This function returns the body of the sell
+*/
+function makeASale(
+	userId: string, 
+	sellingPrice: number, 
+	itemId: string, 
+	itemType: string, 
+	amountOfSellingItems: number, 
+	onLoad:(success: boolean) => void
+): void {
+	const responsePromise: Promise<Response> = fetch('/api/marketplace/addMarketSale/', {
+		method: 'POST',
+		headers: {
+			'Access-Control-Allow-Origin': '*',
+			'Content-Type': 'application/json',
+			'Access-Control-Allow-Credentials': 'true',
+		},
+		body: JSON.stringify({ sellerId: userId, salePrice: sellingPrice, itemId: itemId, itemType: itemType, amount: amountOfSellingItems}),
+	});
+	responsePromise.then (
+		response => response.json().then(data => {
+			if (response.status !== 201) {
+				console.log(response.status);
+				console.log(data);
```

Comment:
Yeah, in general, we want to have toasts when any of our API calls fail. Since this happens for every API call, we should probably handle this at the API level, rather than passing info around and handling it in different places.

Reply:
The toasts are all in place on the receiving end.